indico/modules/categories/controllers/display.py

```diff
@@ -330,3 +331,161 @@ def _process(self):
                                       session.user,
                                       Event.end_dt >= now_utc())
         return send_file(filename, buf, 'application/atom+xml')
+
+
+class RHCategoryOverview(RHDisplayCategoryBase):
+    """Display the events for a particular day, week or month"""
+
+    def _checkParams(self):
+        RHDisplayCategoryBase._checkParams(self)
+        self.detail = request.args.get('detail', 'event')
+        if self.detail not in ('event', 'session', 'contribution'):
+            raise BadRequest('Invalid detail argument')
+        self.period = request.args.get('period', 'day')
+        if self.period not in ('day', 'month', 'week'):
+            raise BadRequest('Invalid period argument')
+        if 'date' in request.args:
+            try:
+                date = datetime.strptime(request.args['date'], '%Y-%m-%d')
+            except ValueError:
+                raise BadRequest('Invalid date argument')
+        else:
+            date = datetime.now()
+        date = self.category.display_tzinfo.localize(date)
+        date = date.replace(hour=0, minute=0, second=0, microsecond=0)
+        if self.period == 'day':
+            self.start_dt = date
+            self.end_dt = self.start_dt + relativedelta(days=1)
+        elif self.period == 'week':
+            self.start_dt = date - relativedelta(days=date.weekday())
+            self.end_dt = self.start_dt + relativedelta(days=7)
+        elif self.period == 'month':
+            self.start_dt = date + relativedelta(day=1)
+            self.end_dt = self.start_dt + relativedelta(months=1)
+
+    def _process(self):
+        # TODO take in consideration protection and visibility
+        events = (Event.query.with_parent(self.category)
```

Comment:
`.filter(Event.category_chain_overlaps(self.category.id), ~Event.is_deleted)` instead of `with_parent`